app/controllers/announcements_controller.rb

```diff
@@ -0,0 +1,54 @@
+class AnnouncementsController < ApplicationController
+  before_action :set_announcement, only: %i[edit update show destroy]
+
+  load_and_authorize_resource
+
+  def index
+    @episode = Episode.find(params['episode_id'])
+  end
+
+  def show; end
+
+  def new
+    @announcement = Announcement.new
+  end
+
+  def create
+    announcement = Announcement.new(announcement_params)
+    if announcement.save
+      redirect_to episode_announcements_path
+    else
+      redirect_to new_episode_announcement_path
+    end
+  end
+
+  def edit; end
+
+  def update
+    if @announcement.update(announcement_params)
+      redirect_to episode_announcement_path
+    else
+      redirect_to edit_episode_announcement_path
+    end
+  end
+
+  def destroy
+    @announcement.destroy
+
+    redirect_to episode_announcements_path
+  end
+
+  private
+
+  def announcement_param_names
+    %i[title date target_resource image video]
+  end
+
+  def announcement_params
+    params.require(:announcement).permit(announcement_param_names).merge(episode_id: params['episode_id'])
```

Comment:
Is `episode_id` permitted?

Reply:
`params.require(:announcement).permit(announcement_param_names)`

Reply:
Do we really need to permit `path` params?